parts/openshift/openshiftmasterscript.sh

```diff
@@ -3,18 +3,34 @@
 # TODO: /etc/dnsmasq.d/origin-upstream-dns.conf is currently hardcoded; it
 # probably shouldn't be
 
-SERVICE_TYPE=origin
-IMAGE_TYPE="${SERVICE_TYPE}"
-IMAGE_PREFIX="openshift"
-ANSIBLE_DEPLOY_TYPE="origin"
 if [ -f "/etc/sysconfig/atomic-openshift-node" ]; then
 	SERVICE_TYPE=atomic-openshift
+else
+	SERVICE_TYPE=origin
+fi
+VERSION="$(rpm -q $SERVICE_TYPE --queryformat %{VERSION})"
+
+if [ -f "/etc/sysconfig/atomic-openshift-node" ]; then
 	ANSIBLE_DEPLOY_TYPE="openshift-enterprise"
 	IMAGE_TYPE=ose
 	IMAGE_PREFIX="registry.access.redhat.com/openshift3"
+	ANSIBLE_CONTAINER_VERSION="v${VERSION}"
+	PROMETHEUS_VERSION="v${VERSION}"
+	PROMETHEUS_EXPORTER_VERSION="v${VERSION}"
+	PROMETHEUS_ALERT_BUFFER_VERSION="v${VERSION}"
+	PROMETHEUS_ALERT_MANAGER_VERSION="v${VERSION}"
+else
+	ANSIBLE_DEPLOY_TYPE="origin"
+	IMAGE_TYPE="${SERVICE_TYPE}"
+	IMAGE_PREFIX="openshift"
+	# FIXME: These versions are set to deal with differences in how Origin and OCP
```

Comment:
Topic for discussion in the next arch call @pweil- @jim-minter ?

Reply:
I was discussing this on IRC with maxamillion and didn't update the PR - I'm of the opinion that this can/should be removed from the PR before it merges.

Reply:
(at least all the new PROMETHEUS_VERSION bits - the existing ANSIBLE_DEPLOY_TYPE/IMAGE_PREFIX stuff is required and will end up getting baked into the image one day)

Reply:
What specifically should be removed?